packages/jsapi-utils/src/useTable.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Column, Row, Table } from '@deephaven/jsapi-shim';
+import Log from '@deephaven/log';
+import useTableListener from './useTableListener';
+
+const log = Log.module('useTable');
+
+const useTable = (
+  table: Table | undefined,
+  columnNames: string[],
+  firstRow: number,
+  lastRow: number
+): {
+  columns: Column[] | undefined;
+  data: unknown[][];
+  error: Error | null;
+  setViewport: (viewport: [number, number]) => void;
+} => {
+  const [data, setData] = useState<unknown[][]>([]);
+  const [viewport, setViewport] = useState([firstRow, lastRow]);
```

Comment:
I don't think this is right... we shouldn't need to return `setViewport`, we're already passing in the viewport `firstRow, lastRow, columnNames`. Should instead just do `const viewport = { top, bottom, columnNames }` and then base the useEffects off of that.
Otherwise we're not really getting hook behaviour - we can change the values of firstRow/lastRow and it won't have an effect, we need to call `setViewport` separately. Instead just bake it into the params.